src/postgres/contrib/yb_pg_metrics/yb_pg_metrics.c

```diff
@@ -528,6 +534,29 @@ ybpgm_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
   {
     instr_time start;
     instr_time end;
+    statementType type;
+
+    if (IsA(pstmt->utilityStmt, TransactionStmt)) {
```

Comment:
I suggest moving the getting `type` logic to a separate function.

Reply:
Done! Addressed all the other recommended changes as well.